jsapp/js/components/permissions/permConstants.ts

```diff
@@ -81,20 +81,20 @@ type CheckboxNameRegular =
   | 'submissionsEdit'
   | 'submissionsValidate'
   | 'submissionsDelete';
-/** Checkboxes for partial permissions (the counterparts). */
-export type CheckboxNamePartial =
-  | 'submissionsViewPartial'
-  | 'submissionsEditPartial'
-  | 'submissionsValidatePartial'
-  | 'submissionsDeletePartial';
-/** All checkboxes combined. */
-export type CheckboxNameAll = CheckboxNameRegular | CheckboxNamePartial;
-/** List of usernames for a partial permission checkbox. */
-export type CheckboxNameListPartial =
-  | 'submissionsViewPartialUsers'
-  | 'submissionsEditPartialUsers'
-  | 'submissionsDeletePartialUsers'
-  | 'submissionsValidatePartialUsers';
+/** Names of checkboxes for partial permissions (the counterparts). */
+export type CheckboxNamePartialByUsers =
+  | 'submissionsViewPartialByUsers'
+  | 'submissionsEditPartialByUsers'
+  | 'submissionsValidatePartialByUsers'
+  | 'submissionsDeletePartialByUsers';
+/** All checkboxes names combined. */
+export type CheckboxNameAll = CheckboxNameRegular | CheckboxNamePartialByUsers;
+/** Name of lists of usernames for a partial permissions checkboxes. */
+export type PartialByUsersListName =
+  | 'submissionsViewPartialByUsersList'
+  | 'submissionsEditPartialByUsersList'
+  | 'submissionsDeletePartialByUsersList'
+  | 'submissionsValidatePartialByUsersList';
```

Comment:
If you turned `CheckboxNamePartialByUsers` and `PartialByUsersListName` into `enum`s, it looks like you [could delete](https://github.com/kobotoolbox/kpi/pull/4731/files#diff-16bcdd306655aaf1598bac0a5f0932c51c2f9bad9572e7fc721aa749b170f960R199-R230) `getPartialByUsersListName()` and `getPartialByUsersCheckboxName()`. (A frozen `const` object would also work, but at that point it's just an `enum` with more steps.)

Reply:
Nice idea, I will try doing it on `feature-extend-partial-perms--add-partial-by-responses` as I think I have more of these functions there already